src/libstd/memchr.rs

```diff
@@ -0,0 +1,339 @@
+/* Initial implementation taken from Andrew Gallant's rust-memchr
```

Comment:
I've added the rust copyright header, as well as rust-memchrs license header.

@bluss I'd be glad to put your name in as well, what should I put in? It's not in the rust-memchr copyright notice.